src/datatypes/src/schema/column_schema.rs

```diff
@@ -262,6 +262,14 @@ impl ColumnSchema {
         );
         Ok(self)
     }
+
+    pub fn set_fulltext_options(&mut self, options: FulltextOptions) -> Result<()> {
+        self.metadata.insert(
```

Comment:
Maybe you can add a check whether the column type is string: `self.data_type == ConcreteDataType::string_datatype()`, and also change the `with_fulltext_options` above